documentation/modules/ROOT/pages/06-quarkus-reactive.adoc

```diff
@@ -0,0 +1,253 @@
+    private String description;
+
+    // generate getters/setters
+----
+copyToClipboard::qext-quarkusrpb-beer-pojo[]
+
+[#quarkusrpb-create-beer-client]
+==== Create Rest Client to Punk Beer API
+
+Add new Java file in `$PROJECT_HOME/src/main/java/com/example` called `BeerGateway` with the following contents:
+
+[#qext-quarkusrpb-beer-client]
+[source,java,subs="+macros,+attributes"]
+----
+package com.example;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+@Path("/v2")
+@RegisterRestClient
+public interface BeerGateway {
+
+    @GET
+    @Path("/beers")
+    @Produces(MediaType.APPLICATION_JSON)
+    CompletionStage<List<Beer>> getBeers(@QueryParam("page") int page);
+    
+}
+----
+copyToClipboard::qext-quarkusrpb-beer-client[]
+
+[#quarkusrpb-configure-props]
+===== Edit the application.properties
+
+Add the following properties to pass:[$PROJECT_HOME]/src/main/resources/application.properties:
+
+[#qext-quarkusrpb-update-props]
+[source,config,subs="+macros,+attributes"]
+----
+com.example.BeerGateway/mp-rest/url=https://api.punkapi.com
+----
+copyToClipboard::qext-quarkusrpb-update-props[]
+
+[#quarkusrpb-beer-reactive-service]
+==== Create Beer Punk API Reactive Service
+
+Let's create the service that uses `BeerGateway` to retrieve beer information by page and reactively emits each of the beer information.
+
+Add new Java file in `$PROJECT_HOME/src/main/java/com/example` called `BeerService` with the following contents:
+
+[#qext-quarkusrpb-beer-service]
+[source,java,subs="+macros,+attributes"]
+----
+package com.example;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import io.reactivex.Flowable;
+
+
+@ApplicationScoped
+public class BeerService {
+
+    ExecutorService executor = Executors.newSingleThreadExecutor(); // <1>
+
+    @RestClient
+    BeerGateway beerGateway;
+
+    public Flowable<Beer> beers() {
+        Flowable<List<Beer>> beerStream = Flowable.generate(() -> 1, (page, emitter) -> { // <2>
+
+            beerGateway.getBeers(page).thenAcceptAsync(beers -> {
+                
+                    if (beers.isEmpty()) {
+                        emitter.onComplete(); // <3>
+                    } else {
+                        emitter.onNext(beers); // <4>
+                    }
+                
+                }, executor);
+
+                return page + 1; // <5>
+            });
+
+            return beerStream.flatMap(Flowable::fromIterable); // <6>
+    }
+
+}
+----
+copyToClipboard::qext-quarkusrpb-beer-service[]
+
+<1> Creates a single thread executor for getting beers by page
```

Comment:
Should you mention that this executor is only there to serialized calls, not as a way to do multi-threading?